Use flow-maven-plugin

SomeoneToIgnore opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/beverage-starter-flow/125)
<!-- Reviewable:end -->

gilberto-torrezan commented:
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[pom.xml, line 138 at r1](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125#-Kz46m52HAh796uwMRsQ:-Kz46m52HAh796uwMRsR:b-zb5xp8) ([raw file](https://github.com/vaadin/beverage-starter-flow/blob/5db84d07be339bccb141ff3b41dc24659135b99d/pom.xml#L138)):*
> ```XML
>         <dependency>
>             <groupId>com.vaadin</groupId>
>             <artifactId>flow-generated-components</artifactId>
> ```

Not related to your patch, but lots of dependencies here could be excluded - for instance, you don't need to include the generated components since you include the flow-components. And that applies for the flow client and server as well.

Just mentioned that because the addition of the next dependencies made the pom.xml really long.

---

*[pom.xml, line 200 at r1](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125#-Kz46I0ZWGAPwAKFU3Am:-Kz46I0_mEOLc-pSLWML:bdk7ku9) ([raw file](https://github.com/vaadin/beverage-starter-flow/blob/5db84d07be339bccb141ff3b41dc24659135b99d/pom.xml#L200)):*
> ```XML
>                 <configuration>
>                     <archive>
>                         <manifestEntries>
> ```

Why is this needed?

---

*[pom.xml, line 264 at r1](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125#-Kz46AOpXNa6EtQ0Gg-D:-Kz46AOpXNa6EtQ0Gg-E:b-u83yi8) ([raw file](https://github.com/vaadin/beverage-starter-flow/blob/5db84d07be339bccb141ff3b41dc24659135b99d/pom.xml#L264)):*
> ```XML
>                         <configuration>
>                             <archive>
>                                 <manifestEntries>
> ```

Why is this needed (again)?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125)*
<!-- Sent from Reviewable.io -->

SomeoneToIgnore commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[pom.xml, line 200 at r1](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125#-Kz46I0ZWGAPwAKFU3Am:-Kz47iFtTIn2Zou4TL5-:b-xmijp9) ([raw file](https://github.com/vaadin/beverage-starter-flow/blob/5db84d07be339bccb141ff3b41dc24659135b99d/pom.xml#L200)):*
<details><summary><i>Previously, gilberto-torrezan (Gilberto Torrezan) wrote…</i></summary><blockquote>

Why is this needed?
</blockquote></details>

To get beautiful MANIFEST.MF entries:

```
Manifest-Version: 1.0
Implementation-Title: Beverage Buddy App Starter for Vaadin Flow
Implementation-Version: 1.0.0.alpha10
<....>
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125)*
<!-- Sent from Reviewable.io -->

gilberto-torrezan commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[pom.xml, line 200 at r1](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125#-Kz46I0ZWGAPwAKFU3Am:-Kz48h2stQCagbgZ07Cz:bxw7g02) ([raw file](https://github.com/vaadin/beverage-starter-flow/blob/5db84d07be339bccb141ff3b41dc24659135b99d/pom.xml#L200)):*
<details><summary><i>Previously, SomeoneToIgnore (Kirill Bulatov) wrote…</i></summary><blockquote>

To get beautiful MANIFEST.MF entries:

```
Manifest-Version: 1.0
Implementation-Title: Beverage Buddy App Starter for Vaadin Flow
Implementation-Version: 1.0.0.alpha10
<....>
```
</blockquote></details>

Right, but... where is this visible? And why do we need beautiful manifest entries?

I'm asking about that just to find ways to make the pom.xml a bit smaller.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125)*
<!-- Sent from Reviewable.io -->

SomeoneToIgnore commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[pom.xml, line 138 at r1](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125#-Kz46m52HAh796uwMRsQ:-Kz47xjtluLj0LIgJtzy:b-rdaznt) ([raw file](https://github.com/vaadin/beverage-starter-flow/blob/5db84d07be339bccb141ff3b41dc24659135b99d/pom.xml#L138)):*
<details><summary><i>Previously, gilberto-torrezan (Gilberto Torrezan) wrote…</i></summary><blockquote>

Not related to your patch, but lots of dependencies here could be excluded - for instance, you don't need to include the generated components since you include the flow-components. And that applies for the flow client and server as well.

Just mentioned that because the addition of the next dependencies made the pom.xml really long.
</blockquote></details>

Makes sense, removed a few.

---

*[pom.xml, line 200 at r1](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125#-Kz46I0ZWGAPwAKFU3Am:-Kz49DA_RJY1LHNhNqjF:b-ucu916) ([raw file](https://github.com/vaadin/beverage-starter-flow/blob/5db84d07be339bccb141ff3b41dc24659135b99d/pom.xml#L200)):*
<details><summary><i>Previously, gilberto-torrezan (Gilberto Torrezan) wrote…</i></summary><blockquote>

Right, but... where is this visible? And why do we need beautiful manifest entries?

I'm asking about that just to find ways to make the pom.xml a bit smaller.
</blockquote></details>

It's visible in the manifest of the war which is used.. idk where. Most probably Tomcat and others are able to display application name from the manifest of no other sources provided.

Have no strong opinion on having this, removed it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125)*
<!-- Sent from Reviewable.io -->

gilberto-torrezan commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[pom.xml, line 21 at r2](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125#-Kz4B8JvpKxDOmCagAzI:-Kz4B8JvpKxDOmCagAzJ:b-aqzaax) ([raw file](https://github.com/vaadin/beverage-starter-flow/blob/1784ac8ed46946955fd62a5ce6dffbbfb49356ce/pom.xml#L21)):*
> ```XML
>         <vaadin.project.version>10.0.0-alpha6</vaadin.project.version>
>         <!-- Needed for production mode -->
>         <flow.project.version>1.0-SNAPSHOT</flow.project.version>
> ```

Now I'm not sure if this should use the SNAPSHOT or the released version (on master). Need to ask the elders.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125)*
<!-- Sent from Reviewable.io -->

SomeoneToIgnore commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[pom.xml, line 21 at r2](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125#-Kz4B8JvpKxDOmCagAzI:-Kz4BQgirtzKLqNkIKyW:bqzowzl) ([raw file](https://github.com/vaadin/beverage-starter-flow/blob/1784ac8ed46946955fd62a5ce6dffbbfb49356ce/pom.xml#L21)):*
<details><summary><i>Previously, gilberto-torrezan (Gilberto Torrezan) wrote…</i></summary><blockquote>

Now I'm not sure if this should use the SNAPSHOT or the released version (on master). Need to ask the elders.
</blockquote></details>

Right, that's my bad, fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125)*
<!-- Sent from Reviewable.io -->

gilberto-torrezan commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/beverage-starter-flow/125)*
<!-- Sent from Reviewable.io -->
